### DISCUSSION TITLE
Contour in WSL VS code

### DISCUSSION THREAD

**Discussion by montyvesselinov** [contour-terminal/contour#639] [General]
Is it possible to make contour  a default terminal under WSL VS codel?!

**Comment by christianparpart** [comment]
@jerch probably knows what you are asking here. I'm not knowledgeable wrt VS code and related ecosystem.
But I'd say as external terminal console, that should be possible and if not, I'd love to know how I can fix that :)

Do you use VScode natively from within Windows and want to spawn a WSL hosted terminal, such as Contour? Or is even your VS code running from within WSL? (I am assuming we are talking about WSL2 here, right?)

**Comment by jerch** [comment]
Nope, no clue what "default terminal under WSL VS code" exactly means, some sort of bridging WSL pty directly to contour running in windows? Not sure if thats possible, I left all windows based dev short before WSL2 came out, by that time the pty fd could not be mapped directly, just through conpty api. This prolly changed to the better now?

Also no idea, how vscode would play into this - it is an electron app, thus unless they hacked into the chrome render core libs, embedding  a totally forgeign process window like contour doesnt seem likely to me.

**Comment by WSLUser** [comment]
Probably referring to the Win11 Defterm work that the Windows Terminal team did that allows you to make Windows Terminal the default terminal (instead of say running CMD or Powershell). There's supposed to be docs into making a different terminal the default terminal you run available though. Xterm.js isn't really an applicable terminal for that since I'm not sure who uses it outside of VS Code or how you would even make it work outside of VS Code. But Contour can definitely take steps to become the default terminal but needs to be sure it allows users to choose and not just explicitly become the default. I think it's part of the code that needs to be added that Contour will need to import in.

**Reply by christianparpart** [reply]
"Defterm"? as in "default terminal"? We've a ticket for that already and it's also definitely about to come. Just not as of now (unless someone volunteers to implement it!). My private time is [sadly] short. So I try to care for  the bigger picture for now. But if no one will pick it up before me, I'll hopefully have time for that this year (I have personal interest in that myself :) ).
